### DISCUSSION TITLE
PHOTOPRISM_AUTO_IMPORT not working in a sub-directory on a shared domain

### DISCUSSION THREAD

**Discussion by chain710** [photoprism/photoprism#2386] [Q&A]
set env PHOTOPRISM_AUTO_IMPORT = 30 & PHOTOPRISM_AUTO_INDEX = 30

using PhotoSync(ios) to upload photo to http://domain/photoprism/import

wait for ~15 min

photo still not imported yet

![image](https://user-images.githubusercontent.com/486539/171213601-440b511e-5215-47fb-93dd-0627d41c1b8a.png)

**Comment by lastzero** [comment]
Do you have server logs, ideally with debug mode enabled? This is just the metadata of the file. Without knowing more, I can only make wild guesses that probably won't help you. For example, it appears that you don't have PhotoPrism installed in the root directory, and we haven't tested that edge case in combination with automatic indexing. So it could be a problem, although I don't know why. You should also report any changes to the configuration, especially workers, worker intervals, and anything else related to indexing.

**Reply by chain710** [reply]
Enabled debug log and no “auto import” log found. I guess WebDAV upload did not trigger ShouldImport correctly.

**Reply by lastzero** [reply]
Did you restart your instance for changes to take effect? What is the output of the "photoprism config" command?

**Reply by chain710** [reply]
yes, I restarted the pod( btw, I deployed the instance on microk8s)

and photoprism config tells the same value 30

**Reply by graciousgrey** [reply]
As PhotoSync is using WebDAV (in case you are using the PhotoPrism target), auto-import should work in general.
The default value for auto-import is 180 the default for auto-index is 300. Try to use those. 
30 Seconds is very short, also there should be different values for auto-import and auto-index so that they do not interfere. 

Just to be sure that you configured PhotoSync correctly: Does importing work when you trigger it using the Web UI (https://docs.photoprism.app/user-guide/library/import/) ? Do you have a subfolder selected in the Web UI?

**Reply by chain710** [reply]
- I tried: default values(300), 2 min(120), 1 min(60), half min(30), and none of those work
- importing work when I click the import button on Web UI

**Reply by chain710** [reply]
bug found:  in webdav.go
```go
// should: router.BasePath() == conf.BaseUri(WebDAVOriginals)
	if router.BasePath() == WebDAVOriginals {
		auto.ShouldIndex()
	} else if router.BasePath() == WebDAVImport {
		auto.ShouldImport()
	}
```

of course other WebDAVOriginals, WebDAVImport refs should be fixed

**Reply by lastzero** [reply]
Seems that my guess wasn't bad at all!?

The first step would be to better document known issues. I should note that support for running PhotoPrism in a subpath is rather experimental. A few people asked for it and we ended up implementing it while we were working on the router anyway, but without testing extra features like automatic import.

Please note that the PWA will also not work if it is not installed in the root path:

- https://docs.photoprism.app/getting-started/faq/#can-i-install-photoprism-in-a-sub-directory-on-a-shared-domain

**Comment by lastzero** [comment]
**[Getting Started FAQ has been updated:](https://docs.photoprism.app/getting-started/faq/#can-i-install-photoprism-in-a-sub-directory-on-a-shared-domain)**

### Can I install PhotoPrism in a sub-directory on a shared domain?

This is possible if you run it [behind a reverse proxy](proxies/traefik.md), though *experimental and not extensively tested*. Depending on [the resources available to us](https://photoprism.app/membership), we [work to resolve known issues](https://link.photoprism.app/roadmap) over time.

To give specific examples: (a) For our [Progressive Web App (PWA)](../user-guide/pwa.md) to work as intended, [the service worker](https://web.dev/learn/pwa/service-workers/) should be installed in the root directory. (b) Due to the way Web browsers function, sharing a domain with other services [can affect the performance](https://docs.pushtechnology.com/cloud/latest/manual/html/designguide/solution/support/connection_limitations.html) and [security](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) of all the apps installed. (c) The length of [sharing and other links](../user-guide/share/index.md) increases and may not match the documentation. (d) There may be problems with WebDAV sharing and [automatic indexing/importing](https://github.com/photoprism/photoprism/discussions/2386).

**Reply by lastzero** [reply]
I also created a GitHub issue to help other users find the information and perhaps find contributors to take care of the remaining issues:

- #2391

**Reply by chain710** [reply]
auto import/index issue should be easily fixed ？

I hack the code and it worked as intended on my microk8s

**Reply by lastzero** [reply]
I assume so, but we already have a very long list of issues that need to be addressed in the short term, and I have no idea when we'll release the next time. Sounds like you can solve the problem by yourself, as long as no general solution has been released?

**Reply by lastzero** [reply]
Feel free to submit a pull request to document the changes that work for you as this will save us time :+1:

**Reply by chain710** [reply]
Ok, will submit pr later, but still not figure out how to build amd64 image yet
I noticed someone asked about docker image [https://github.com/photoprism/photoprism/discussions/2282](here), but not answered yet
Can someone elaborate? Many thanks

**Reply by lastzero** [reply]
Thank you very much! How to set up the build/development environment is described in the Developer Guide: https://docs.photoprism.app/developer-guide/setup/

Our Makefile includes `make docker-local` targets to build local container images tagged `photoprism/photoprism:local`. Unlike the builds for the official preview and release images, they won't be automatically pushed to Docker Hub. However, you can re-tag and push them manually if needed, for example to a private Docker registry.

https://github.com/photoprism/photoprism/blob/e1f14b7ec2076c4ec6bcd4bc52b54d3504c932e2/Makefile#L395
